src/main/java/com/example/clone/service/FriendDetails.java

```diff
@@ -0,0 +1,89 @@
+package com.example.clone.service;
+
+import com.example.clone.model.UserInfo;
+import com.example.clone.repository.UserInfoRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Service
+public class FriendDetails {
+
+    @Autowired
+    private UserInfoRepository userRepository;
+
+    private final GetAuth getAuth;
+
+    @Autowired
+    public FriendDetails(UserInfoRepository userRepository, GetAuth getAuth){
+        this.userRepository = userRepository;
+        this.getAuth = getAuth;
+    }
+    public void addFriend(int friendId) throws Exception {
+
+
+        int currentUser = getAuth.getCurrentUser().getId();
+
+        UserInfo user = userRepository.findById(currentUser)
+                .orElseThrow(() -> new Exception("User not found"));
+        UserInfo friend = userRepository.findById(friendId)
+                .orElseThrow(() -> new Exception("Friend not found"));
+
+
+        if (user.getFriends().contains(friend)) {
+            throw new Exception("They are already friends");
```

Comment:
Avoid using exceptions for flow control. This can be handled by returning the value from the method.
Exceptions must be used only in exceptional situations, if the error is part of business logic then it must be handled without exceptions.
Exceptions are ~100 times slower than the regular app flow.

Reply:
I've implemented response classes for each service, I'm unsure if that's the best way to do it but I can remove them/change it up if it's not.